### ISSUE TITLE
Numbered list wrong numeration after increase + decrease indent

### ISSUE THREAD

**Issue by Alspb** [singerdmx/flutter-quill#1974]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### Flutter Quill version

master

### Steps to reproduce

1. Increase the numbered list indent 3 times, until the numbering style is the same again.
2. For the last list item decrease its indent 3 times.

### Expected results

The last item numeration start from 1.

### Actual results

The last item numeration start from 2.

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

![Numbered list numeration bug](https://github.com/singerdmx/flutter-quill/assets/73047043/dbbf1a8d-82c2-4679-8033-d4f99b9c34f7)

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by AtlasAutocode** [comment]
Part of the confusion might be that indents are limited to 5 levels (no idea why).
Ordered lists seem to have only 3 levels so it repeats the sequence.
When you indent 3 times you get back to '1.' However, this is not the same as the un-indented '1.'
So, the next outdented list item will be '2.'. That is correct (but not what you want!)

It appears that the first list item is implicitly set to 1 even if it is shown as an indented value.
For example: start a list and indent to give the first item 'a: text'. Enter another list item, outdent it and it will be '2. next'.

I am not familiar with the code, but I guess that there is an overlap between indentation of text and indentation of list items.
Am I correct that you want to create a normal list, but have it indented?

I see several approaches:
1. Resolve the confusion between indentation and list-depth
2. Have 5 levels of list indentation:  1,  a,  i, and then what?
3. I am sure there will be some other way....

**Comment by AtlasAutocode** [comment]
I was thinking that it might be more logical if indentation of the first item could be handled as an indentation of the whole list.
The first item would then always be '1.' and you would not be able to 'outdent' the second item. (Avoids the issue in your screen shot)
Indentation of subsequent items would then follow the expected pattern 'a', 'i' ...
Allowing the user to pick a different starting label would be tricky. But word processors allow starting a list at '4' (for example). I often have to do this when continuing a list after an interruption.

Of course, it is easy for me to suggest this, as I have no idea if this can be done...

**Comment by CatHood0** [comment]
> Also, delta to HTML converter treats such cases as if "text" is not indented too

This could be because the way Delta handles its lists is different from HTML which makes the conversion really complicated and has to be generalized.

**Comment by CatHood0** [comment]
> Of course, it is easy for me to suggest this, as I have no idea if this can be done...

Literally me

**Comment by CatHood0** [comment]
I guess we need to add more variables that reference the list indentation level to fix this more easily. I did this in `flutter_quill_to_pdf` and it solved exactly the same problem as this one since it only updates the list number depending on the indentation it has and resets it the same way.

**Comment by AtlasAutocode** [comment]
Looks good - go for it.

My only question is: What if someone in the future changes the maximum allowed indentation level?
Could you use numberIndentList[x] where the size of the array is a constant for max indentation also referenced in limiting indentation level.

That may also simplify the code a bit.

**Comment by CatHood0** [comment]
@AtlasAutocode  i fix it. I just need to add this condition to clear correctly the indents:
```dart
      if (prevNodeOl && attrs[Attribute.list.key] != Attribute.ol || attrs.isEmpty) {
        clearIndents = true;
      }
```

This simple condition clear correctly the indent and fix the numeration of the list.

**Comment by AtlasAutocode** [comment]
There is a programming concept called 'Extreme Programming' which notes that: 'bugs are always found in code that was written'.
Therefore: 'Write the least amount of code that solves the problem'
2 lines is good!

**Comment by CatHood0** [comment]
> But how does it solve the issue

It doesn't fix it directly. I mean that while creating that Feature, I happened to find how to fix the error. It was just necessary to add an additional check in `_buildChildren` of `raw_editor_state` so that it would be reset if necessary

**Comment by CatHood0** [comment]
Using `customLeadingBlockBuilder` you can use `config.getIndexNumberByIndent;` to get the correct index for ordered lists (you can assume that `config` is the param passed by  `customLeadingBlockBuilder`  that contains this method).

**Comment by CatHood0** [comment]
> Do you mean that you have

Yeah, that's the result that i getting. But, if you have the lastest version of the package, and this persist, please show a video or steps to reproduce it.

**Comment by CatHood0** [comment]
I'll take a look.
